spark/src/main/scala/org/apache/spark/sql/delta/schema/SchemaMergingUtils.scala

```diff
@@ -192,7 +194,8 @@ object SchemaMergingUtils {
       keepExistingType: Boolean,
       typeWideningMode: TypeWideningMode,
       caseSensitive: Boolean,
-      allowOverride: Boolean): DataType = {
+      allowOverride: Boolean,
+      overrideMetadata: Boolean = false): DataType = {
```

Comment:
nit: for consistency, don't provide a default value here, but pass it when calling `mergeDataTypes` in `mergeSchemas`
```suggestion
      overrideMetadata: Boolean): DataType = {
```